plugins/parodos/src/components/notification/NotificationsList.tsx

```diff
@@ -0,0 +1,276 @@
+    maxWidth: theme.breakpoints.values.xl,
+  },
+  root: {
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(2),
+  },
+  tbody: {
+    '& tr:last-child': {
+      borderBottom: `1px solid ${theme.palette.grey.A100}`,
+    },
+  },
+}));
+
+type Action = 'DELETE' | 'ARCHIVE';
+
+type NotificationListProps = Pick<
+  NotificationsSlice,
+  | 'notifications'
+  | 'fetchNotifications'
+  | 'deleteNotification'
+  | 'notificationsLoading'
+  | 'setNotificationState'
+> & { fetch: FetchApi['fetch'] };
+
+export function NotificationList({
+  notifications,
+  fetchNotifications,
+  deleteNotification,
+  notificationsLoading: loading,
+  setNotificationState,
+  fetch,
+}: NotificationListProps): JSX.Element {
+  const styles = useStyles();
+  const [selectedNotifications, setSelectedNotifications] = useState<
+    NotificationContent[]
+  >([]);
+  const [dialogOpen, setOpenDialog] = useState(false);
+  const [action, setAction] = useState<Action>('ARCHIVE');
+  const [showAlert, setShowAlert] = useState(false);
+
+  const [notificationFilter, setNotificationFilter] =
+    useState<NotificationState>('ALL');
+
+  const [page, setPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(10);
+
+  const errorApi = useApi(errorApiRef);
+
+  const handleChangePage = (
+    _event: MouseEvent<HTMLButtonElement> | null,
+    newPage: number,
+  ) => {
+    setPage(newPage);
+  };
+
+  useEffect(() => {
+    fetchNotifications({
+      filter: notificationFilter,
+      page,
+      rowsPerPage,
+      fetch,
+    });
+  }, [notificationFilter, page, rowsPerPage, fetchNotifications, fetch]);
+
+  const handleChangeRowsPerPage = (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  const filterChangeHandler = (filter: SelectedItems) => {
+    setNotificationFilter(filter as NotificationState);
+    setPage(0);
+  };
+
+  const deleteNotificationsHandler = useCallback(
+    (e: MouseEvent<HTMLButtonElement>) => {
+      e.stopPropagation();
+
+      setAction('DELETE');
+      setOpenDialog(true);
+    },
+    [],
+  );
+
+  const dialogYesHandler = useCallback(
+    async (e: MouseEvent<HTMLButtonElement>) => {
+      e.stopPropagation();
+
+      if (loading) {
+        return;
+      }
+
+      try {
+        for (const notification of selectedNotifications) {
+          if (action === 'DELETE') {
+            await deleteNotification({ fetch, id: notification.id });
+          } else {
+            await setNotificationState({
+              fetch,
+              id: notification.id,
+              newState: 'ARCHIVE',
+            });
+          }
+
+          setOpenDialog(false);
+          setShowAlert(true);
+
+          setSelectedNotifications([]);
+
+          await fetchNotifications({
+            fetch,
+            filter: notificationFilter,
+            page,
+            rowsPerPage,
+          });
```

Comment:
Should we recalculate current page? For example if we on the last page and delete more than `rowsPerPage` items.

Reply:
The reason I do a fetch each time is because the query to the API is paged, so we might only have 10 records in memory but the total records in the database is 30.  If we delete 7, we need to call fetch to get a new page full of records.  

That is a good point about the current page.  I've updated the code